EasyGYM/UI/Controllers/Core/LibraryViewController.swift

```diff
@@ -14,221 +14,143 @@ import FirebaseFirestore
 
 class LibraryViewController: UIViewController {
     
+    var viewModel: LibraryViewModel!
+    
     @IBOutlet weak var searchBar: UISearchBar!
     @IBOutlet weak var workoutTableView: UITableView!
-    var favoriteWorkoutIDs: Set<String> = []
-    
     var workouts: [Workout] = []           // Array to hold all workouts fetched from Firestore
     var filteredSearchWorkouts: [Workout] = []   // Array to hold filtered workouts based on search query
     var filteredWorkouts: [Workout] = []
     var isSearching = false                // Flag to track if user is currently searching
-    var isFiltering = false
+    var isFiltering = false                // Flag to track if filtering on
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
+        if viewModel == nil {
```

Comment:
this should not be done. 
Set the viewModel when LibraryViewController is being called